clarity/utils/file_io.py

```diff
@@ -14,3 +25,131 @@ def write_jsonl(filename: str, records: list) -> None:
     with open(filename, "a", encoding="utf-8") as fp:
         for record in records:
             fp.write(json.dumps(record) + "\n")
+
+
+# Functions for reading and writing audio signals.
+#
+# - Signals are passed and returns as numpy arrays of floats of shape
+#   (n_samples, n_channels). Mono signals can be passed as (n_samples, 1)
+#   but are returned as (n_samples,) which is more convenient for most purposes.
+#   Floating point and 16 bit integer file formats are supported.
+# - Signals are assumed to be in the range [-1.0 to 1.0) and will be scaled to
+#   and from -32768 to 32767 when writing to or reading from 16 bit files.
+#   On writing to 16 bit files, overflow can be set to silently wrap or to raise
+#   an error.
+# - Any sample rate can be used for writing and floating points values are
+#   rounded to integers. For reading, the 'expected sample rate' is supplied and
+#   the file is resample if necessary or can be set to throw an error. Again,
+#   this is more convenient for most purposes.
+
+
+def write_signal(
+    filename: str | Path,
+    signal: ndarray,
+    sample_rate: float,
+    floating_point: bool = True,
+    strict: bool = False,
```

Comment:
I realise it was only a test in which it was used but was wondering if it might be of general use to have an option in `write_signal()` and `read_signal()` to specify how much of a sample to read/write?

Reply:
Yep. It's in read_signal().  I think it's less well motivated for write signal as it's easy to use numpy slicing to pass the segment directly. For read_signal the offset and n_samples parameter get passed right down into the SoundFile call and so can potentially speed things up, i.e. avoid reading a complete signal and then truncating.